edx_django_utils/plugins/README.rst

```diff
@@ -0,0 +1,248 @@
+Django App Plugins
+==================
+
+Provides functionality to enable improved plugin support of Django apps.
+
+Once a Django project is enhanced with this functionality, any participating
+Django app (a.k.a. Plugin App) that is PIP-installed on the system is
+automatically included in the Django project's INSTALLED_APPS list. In addition,
+the participating Django app's URLs and Settings are automatically recognized by
+the Django project. Furthermore, the Plugin Signals feature allows Plugin Apps
+to shift their dependencies on Django Signal Senders from code-time to runtime.
+
+While Django+Python already support dynamic installation of components/apps,
+they do not have out-of-the-box support for plugin apps that auto-install
+into a containing Django project.
+
+This Django App Plugin functionality allows for Django-framework code to be
+encapsulated within each Django app, rather than having a monolith Project that
+is aware of the details of its Django apps. It is motivated by the following
+design principles:
+
+* Single Responsibility Principle, which says "a class or module should have
+  one, and only one, reason to change." When code related to a single Django app
+  changes, there's no reason for its containing project to also change. The
+  encapsulation and modularity resulting from code being co-located with its
+  owning Django app helps prevent "God objects" that have too much responsibility
+  and knowledge of the details.
+
+* Open Closed Principle, which says "software entities should be open for
+  extension, but closed for modification." The edx-platform is extensible via
```

Comment:
Search for `edx-platform` in this doc and see where `The edx-platform` should become `An IDA`.